### ISSUE TITLE
import i18next from 'i18next' does not work with typescript

### ISSUE THREAD

**Issue by scharf** [i18next/i18next-gitbook#101]
It seems that the docu is out of sync with the `i18next` module. TypeSctipt complains about the import as suggested in the docu:

```typescript
import i18next from 'i18next' 
```

The way it works is:
```typescript
import * as i18next from 'i18next';
```

**Comment by jamuhl** [comment]
https://github.com/i18next/i18next/blob/master/test/typescript/nonEsModuleInterop/exposed.test.ts
https://github.com/i18next/i18next/blob/master/test/typescript/init.test.ts

both work depending on ts config -> check above tests

**Comment by scharf** [comment]
Thanks to [a hint](https://github.com/i18next/i18next-xhr-backend/issues/327#issuecomment-537814116) by @rosskevin I found a solution:

Adding the option `"esModuleInterop": true` to `tsconfig.json` the original (suggested) import works again:

```
import i18next from 'i18next' 
```
The point is, that something changed in `i18ext` that broke this import.

I had to add `"esModuleInterop": true`, otherwise I need a [terrible hack to import `i18next-xhr-backend`](https://github.com/i18next/i18next-xhr-backend/issues/327#issuecomment-538051994)

What confused me was that updating `i18next` form version `15.1.1` to `17.0.18` broke my code without me changing anything. That happened before, when I switched from `11.9.1` to `15.0.6` I had to change the import form 

```typescript
import * as i18next from 'i18next';
```

to

```typescript
import i18next from 'i18next';
```

I have lots of dependencies but the `i18n` family of plugins now caused a few times headaches for me and therefore I reported the error (those are the only plugin that broke the imports after an update).

As I said, I did not know about the `esModuleInterop` option which seems to solve some problems.

I have to admit, I am still a bit confused how imports in typescript/javascript work and I have an an old unanswered question at stackoverflow: "[When to use which import style for of CommonJs (node) modules in Typescript?](https://stackoverflow.com/questions/35442039/when-to-use-which-import-style-for-of-commonjs-node-modules-in-typescript)"
